web/components/dashboard/widgets/recent-activity.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect } from "react";
+import Link from "next/link";
+import { observer } from "mobx-react-lite";
+import { History } from "lucide-react";
+// hooks
+import { useDashboard, useUser } from "hooks/store";
+// components
+import { ActivityIcon, ActivityMessage } from "components/core";
+import { RecentActivityEmptyState, WidgetLoader } from "components/dashboard/widgets";
+// ui
+import { Avatar } from "@plane/ui";
+// helpers
+import { calculateTimeAgo } from "helpers/date-time.helper";
+// types
+import { TRecentActivityWidgetResponse } from "@plane/types";
+
+type Props = {
+  dashboardId: string;
+  workspaceSlug: string;
+};
+
+const WIDGET_KEY = "recent_activity";
+
+export const RecentActivityWidget: React.FC<Props> = observer((props) => {
+  const { dashboardId, workspaceSlug } = props;
+  // store hooks
+  const { currentUser } = useUser();
+  // derived values
+  const { fetchWidgetStats, widgetStats: allWidgetStats } = useDashboard();
+  const widgetStats = allWidgetStats?.[workspaceSlug]?.[dashboardId]?.[WIDGET_KEY] as TRecentActivityWidgetResponse[];
+
+  useEffect(() => {
+    if (!widgetStats)
+      fetchWidgetStats(workspaceSlug, dashboardId, {
+        widget_key: WIDGET_KEY,
+      });
+  }, [dashboardId, fetchWidgetStats, widgetStats, workspaceSlug]);
+
+  if (!widgetStats) return <WidgetLoader widgetKey={WIDGET_KEY} />;
+
+  return (
+    <Link
+      href="/profile/activity"
+      className="bg-custom-background-100 rounded-xl border-[0.5px] border-custom-border-200 w-full py-6 hover:shadow-custom-shadow-4xl duration-300"
+    >
+      <div className="flex items-center justify-between gap-2 px-7">
+        <h4 className="text-lg font-semibold text-custom-text-300">My activity</h4>
+      </div>
+      {widgetStats.length > 0 ? (
+        <div className="space-y-6 mt-4 mx-7">
+          {widgetStats.map((activity) => (
+            <div key={activity.id} className="flex gap-5">
+              <div className="flex-shrink-0">
+                {activity.field ? (
```

Comment:
please avoid conditionals inside tsx, It will make it look more readable

Reply:
As there are too many conditions inside tsx, we better break this down into smaller components.

Reply:
This same logic is being used wherever we are rendering activity, I think it'll be better to have a separate PR just to refactor the activity logic.